src/models/library_with_filters.rs

```diff
@@ -57,6 +57,62 @@ pub enum Sort {
     LastWatched,
     Name,
     TimesWatched,
+    Watched,
+    NotWatched,
+}
+
+impl Sort {
+    /// [`Sort`]ing the two given [`LibraryItem`]s for the Library
+    pub fn sort_items(&self, a: &LibraryItem, b: &LibraryItem) -> Ordering {
+        match &self {
+            Sort::LastWatched => b.state.last_watched.cmp(&a.state.last_watched),
+            Sort::TimesWatched => b.state.times_watched.cmp(&a.state.times_watched),
+            // the only difference between the Watched and Not watched sorting
+            // is the ordering of the `a` and `b` items
+            Sort::Watched => Self::sort_watched(a, b),
+            Sort::NotWatched => Self::sort_watched(b, a),
+            Sort::Name => a.name.to_lowercase().cmp(&b.name.to_lowercase()),
+        }
+    }
+
+    /// Helper function that sorts by watched
+    /// The only difference between sorting by Watched and Not Watched is the ordering of `a` and `b`.
+    ///
+    /// Watched items are sorted by:
+    /// 1. Show **watched** items ([`LibraryItem::watched()`]` == true`) and sort them by
+    /// [`LibraryItem.state.last_watched`] descending (first show the most recently watched)
+    /// 2. Show **not watched** items next - [`LibraryItem::watched()`]` == false`
+    /// and sort them by [`LibraryItem::ctime`] ascending
+    ///
+    /// Not watched items are just sorted the exact opposite way:
+    /// 1. Show **not watched** items sorted by `ctime` descending (oldest to newest)
+    /// 2. Show **watched** items sorted by `last_watched` ascending
+    fn sort_watched(a: &LibraryItem, b: &LibraryItem) -> Ordering {
+        let a_is_watched = a.watched();
+        let b_is_watched = b.watched();
+
+        match b_is_watched.cmp(&a_is_watched) {
+            Ordering::Equal => {
+                // doesn't matter if we check a or b
+                if a_is_watched {
+                    match (&a.state.last_watched, &b.state.last_watched) {
+                        (Some(a_last_watched), Some(b_last_watched)) => {
+                            b_last_watched.cmp(a_last_watched)
+                        }
+                        _ => Ordering::Equal,
+                    }
+                } else {
+                    // there should always be ctime for LibraryItems added to the Library
+                    match (&b.ctime, &a.ctime) {
+                        (Some(b_ctime), Some(a_ctime)) => b_ctime.cmp(a_ctime),
+                        // if one or both are missing, we just leave the ordering by default
+                        _ => Ordering::Equal,
+                    }
+                }
+            }
+            ordering => ordering,
+        }
+    }
```

Comment:
Could be simplified to this
```suggestion
            Sort::Watched => b
                .watched()
                .cmp(&a.watched())
                .then(b.state.last_watched.cmp(&a.state.last_watched))
                .then(b.ctime.cmp(&a.ctime)),
            Sort::NotWatched => a
                .watched()
                .cmp(&b.watched())
                .then(b.state.last_watched.cmp(&a.state.last_watched))
                .then(b.ctime.cmp(&a.ctime)),
            Sort::Name => a.name.to_lowercase().cmp(&b.name.to_lowercase()),
        }
    }
```

Reply:
Thank you very much for the suggestion. It works and I've added an additional item to test a watched movie (with last_watched but flagged_watched == 0) and it's ordering.